lib/Support/Check.cpp

```diff
@@ -0,0 +1,81 @@
+//===- Check.cpp - Check test results  ------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+//
+//===----------------------------------------------------------------------===//
+
+#include "Support/Check.h"
+
+static bool isDenorm(float F) {
+  return (std::numeric_limits<float>::denorm_min() <= F &&
+          F < std::numeric_limits<float>::min()) ||
+         (-std::numeric_limits<float>::min() < F &&
+          F <= -std::numeric_limits<float>::denorm_min());
+}
```

Comment:
Is this different from `!std::isnormal()`?

Reply:
Looks like yes, 0.0, inf, and NaN are not normal.
https://en.cppreference.com/w/cpp/numeric/math/isnormal

Reply:
Looks like `std::fpclassify(F) == FP_SUBNORMAL` might be the right thing to do here then (See https://en.cppreference.com/w/cpp/numeric/math/fpclassify). Note that subnormal and denormal are equivalent in this context.